### PR TITLE
Date/Time

### PR DESCRIPTION
**Pull request by nstdio**
http://docs.php.net/manual/ru/book.datetime.php

### CODE REVIEW

**Comment by MWGuy** [comment]
Wow. Good job 👍

File: jphp-runtime/src/php/runtime/common/Messages.java [hunk 1]
```diff
@@ -128,6 +128,9 @@ private Messages(){}
     public final static Item ERR_INVALID_UNICODE_ESCAPE_SEQUENCE = new Item("Invalid UTF-8 codepoint escape sequence");
     public final static Item ERR_INVALID_UNICODE_ESCAPE_SEQUENCE_WITH_REASON = new Item(ERR_INVALID_UNICODE_ESCAPE_SEQUENCE.message + ": %s");
 
+    public final static Item ERR_TIMEZONE_NULL_BYTE = new Item("%s: Timezone must not contain null bytes");
+    public final static Item ERR_WRONG_PARAM_TYPE = new Item("%s expects parameter %d to be %s, %s given");
```
[the remark this answers is not in the document]
Comment (dim-s) [reply jphp-runtime/src/php/runtime/common/Messages.java]: @nstdio only the `ERR_TIMEZONE_NULL_BYTE` constant.

**Comment by dim-s** [comment]
How a user can update timezone data without updating this module? It's possible?

**Comment by dim-s** [comment]
I’m not completely sure, but it might be better to separate this extension into a separate module, like jphp-zend-datetime-ext.

**Comment by nstdio** [comment]
> How a user can update the timezone data without updating this module? It's possible?

I guess its not possible, at least as far as I know this can't be done seamlessly. Currently we store timezone database in most dummy manner, it's just a `JSON` file in resources.

**Pull request merged**